odoo/modules/loading.py

```diff
@@ -25,6 +28,49 @@
 _test_logger = logging.getLogger('odoo.tests')
 
 
+def _get_files_of_kind(kind, package):
+    if kind == 'demo':
+        kind = ['demo_xml', 'demo']
+    elif kind == 'data':
+        kind = ['init_xml', 'update_xml', 'data']
+    if isinstance(kind, str):
+        kind = [kind]
+    files = []
+    for k in kind:
+        for f in package.data[k]:
+            files.append(f)
+            if k.endswith('_xml') and not (k == 'init_xml' and not f.endswith('.xml')):
+                # init_xml, update_xml and demo_xml are deprecated except
+                # for the case of init_xml with csv and sql files as
+                # we can't specify noupdate for those file.
+                correct_key = 'demo' if k.count('demo') else 'data'
+                _logger.warning(
+                    "module %s: key '%s' is deprecated in favor of '%s' for file '%s'.",
+                    package.name, k, correct_key, f
+                )
+    return files
+
+
+def extract_data_models(package):
+    """
+    Extracts **all** models that should be loaded in order to create records from a module's
+    data files.
+    """
+    models = tools.OrderedSet()
+    for path in _get_files_of_kind('data', package):
+        # only take into account xml and csv data files: a lot of this code is copied over from
+        # tools/convert.py for proper file type filtering and opening and reading the files
+        # properly
+        fname, ext = os.path.splitext(os.path.basename(path))
+        ext = ext.lower()
+        if ext == '.xml':
+            with tools.file_open(os.path.join(package.name, path), 'rb') as fd:
+                models |= tools.OrderedSet(etree.parse(fd).getroot().xpath('//record/@model'))
```

Comment:
Wouldn't it be simpler to use `.update` and not have to manually create an orderedset?